opts/port.go

```diff
@@ -94,11 +94,20 @@ func (p *PortOpt) Set(value string) error {
 	} else {
 		// short syntax
 		portConfigs := []swarm.PortConfig{}
-		// We can ignore errors because the format was already validated by ValidatePort
-		ports, portBindings, _ := nat.ParsePortSpecs([]string{value})
+		ports, portBindingMap, err := nat.ParsePortSpecs([]string{value})
+		if err != nil {
+			return err
+		}
+		for _, portBindings := range portBindingMap {
+			for _, portBinding := range portBindings {
+				if portBinding.HostIP != "" {
+					return fmt.Errorf("HostIP is not supported by a service.")
```

Comment:
Since this code isn't in the service package anymore, I think it's wrong to say anything about services. This code could easily be used somewhere else and it would be difficult to notice the error message is wrong.

How about just "HostIP is not supported in %s" or something like that? 

This function is already pretty massive as well. What do you think about making this a separate function?